lib/module.js

```diff
@@ -88,9 +88,18 @@ function readPackage(requestPath) {
 function tryPackage(requestPath, exts) {
   var pkg = readPackage(requestPath);
 
-  if (!pkg) return false;
+  if (!(pkg && pkg.main)) return false;
+
+  var resolved = pkg['_resolved'],
```

Comment:
{
      "name": "commander",
      "version": "2.6.0",
      "description": "the complete solution for node.js command-line programs",
      "keywords": [
        "command",
        "option",
        "parser",
        "prompt"
      ],
      "author": {
        "name": "TJ Holowaychuk",
        "email": "tj@vision-media.ca"
      },
      "license": "MIT",
      "repository": {
        "type": "git",
        "url": "https://github.com/tj/commander.js.git"
      },
      "devDependencies": {
        "should": ">= 0.0.1"
      },
      "scripts": {
        "test": "make test"
      },
      "main": "index",
      "engines": {
        "node": ">= 0.6.x"
      },
      "files": [
        "index.js"
      ],
      "gitHead": "c6807fd154dd3b7ce8756f141f8d3acfcc74be60",
      "bugs": {
        "url": "https://github.com/tj/commander.js/issues"
      },
      "homepage": "https://github.com/tj/commander.js",
      "_id": "commander@2.6.0",
      "_shasum": "9df7e52fb2a0cb0fb89058ee80c3104225f37e1d",
      "_from": "commander@2.6.0",
      "_npmVersion": "2.1.12",
      "_nodeVersion": "0.11.14",
      "_npmUser": {
        "name": "zhiyelee",
        "email": "zhiyelee@gmail.com"
      },
      "maintainers": [
        {
          "name": "tjholowaychuk",
          "email": "tj@vision-media.ca"
        },
        {
          "name": "somekittens",
          "email": "rkoutnik@gmail.com"
        },
        {
          "name": "zhiyelee",
          "email": "zhiyelee@gmail.com"
        },
        {
          "name": "thethomaseffect",
          "email": "thethomaseffect@gmail.com"
        }
      ],
      "dist": {
        "shasum": "9df7e52fb2a0cb0fb89058ee80c3104225f37e1d",
        "tarball": "http://registry.npmjs.org/commander/-/commander-2.6.0.tgz"
      },
      "directories": {},
      "_resolved": "http://registry.npmjs.org/commander/-/commander-2.6.0.tgz",
      "readme": "ERROR: No README data found!"
    }